### DISCUSSION TITLE
JavaScript SDK - Async Storage functions

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#10461] [Feature Requests]
Since i want to use the JavaScript SDK in my mobile app development and using react native, i am limited to use "AsyncStorage".
With the use of the async storage, the calls, well need to be called in an asynchron way.

It would be nice to let IStorage make the functions async and be used with an await.

**Comment by rijkvanzanten** [comment]
afaik you can call `await` on synchronous functions without side-effects, which would allow us to integrate this without breaking the current usage and workflows @joselcvarela

**Reply by joselcvarela** [reply]
Nice effort Rijk 😛
But I believe that would not work since we are relying on storage to get the `access_token` to pass into requests.
That being said, like op mentioned we need to add async/await to storage methods.
@NilsBaumgartner1994 we are open for a PR 😅

**Reply by rijkvanzanten** [reply]
> But I believe that would not work since we are relying on storage to get the access_token to pass into requests.

Makes sense. Though seeing the requests themselves are async, we might be able to await the storage in those method calls? 🤔

**Comment by NilsBaumgartner1994** [comment]
Hi,

Afaik I think you could add await before that. I will get in touch tomorrow with this problem.
Gladly I found a package which temporarily fixes my problem (loading the whole async storage in a dict), but still not very nice.

I created a small client app with themes, a login and menu structure on a Single Code base for Web and native apps (iOS and Android) and using directus sdk.


Beste Grüße,
Nils Baumgartner

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
